warehouse/retry.go

```diff
@@ -64,108 +64,133 @@ func (retryReq *RetryRequest) RetryWHUploads() (response RetryResponse, err erro
 	// Retry request should trigger on these cases.
 	// 1. Either provide the retry interval.
 	// 2. Or provide the List of Upload id's that needs to be re-triggered.
-	uploadsRetried, err := retryReq.retryUploads(sourceIDs)
+	uploadsRetried, err := retryReq.API.warehouseDBHandle.RetryUploads(ctx, retryReq.clausesQuery(sourceIDs)...)
 	if err != nil {
+		err = fmt.Errorf("failed retrying uploads, error: %s", err.Error())
 		return
 	}
 
 	response = RetryResponse{
-		Message:    successMessage(uploadsRetried),
+		Count:      uploadsRetried,
 		StatusCode: 200,
 	}
 	return
 }
 
-func (retryReq *RetryRequest) getSourceIDs() (sourceIDs []string) {
-	sourceIDsByWorkspaceLock.RLock()
-	defer sourceIDsByWorkspaceLock.RUnlock()
-	var ok bool
-	if sourceIDs, ok = sourceIDsByWorkspace[retryReq.WorkspaceID]; !ok {
-		sourceIDs = []string{}
-	}
-	return sourceIDs
-}
-
-func (retryReq *RetryRequest) retryUploads(sourceIDs []string) (rowsAffected int64, err error) {
-	var clausesQuery string
-	var clauses []string
-	var clausesArgs []interface{}
-
-	// Preparing clauses query
-	clauses, clausesArgs = retryReq.whereClauses(sourceIDs)
-	for i, clause := range clauses {
-		clausesQuery = clausesQuery + strings.Replace(clause, retryQueryPlaceHolder, fmt.Sprintf("$%d", i+1), 1)
-		if i != len(clauses)-1 {
-			clausesQuery += " AND "
+func (retryReq *RetryRequest) UploadsToRetry(ctx context.Context) (response RetryResponse, err error) {
+	// Request validation
+	err = retryReq.validateReq()
+	defer func() {
+		if err != nil {
+			retryReq.API.log.Errorf(`WH: Error occurred while fetching upload jobs to retry for workspaceId: %s, sourceId: %s, destinationId: %s with error: %s`,
+				retryReq.WorkspaceID,
+				retryReq.SourceID,
+				retryReq.DestinationID,
+				err.Error(),
+			)
+			response = RetryResponse{
+				Message:    err.Error(),
+				StatusCode: 400,
```

Comment:
Would every outcome from this function be interpreted as `BadRequest` ?

Reply:
Currently, for every error, we are returning 400 for all Grpc calls.

Reply:
Although it doesn't matter what status code we send because cp-router always sends 500 for errors.